R/module_teal_with_splash.R

```diff
@@ -95,15 +106,82 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
           )
         }
       })
-
-      if (!is.reactive(raw_data)) {
-        stop("The delayed loading module has to return a reactive object.")
-      }
       raw_data
     }
 
-    res <- srv_teal(id = "teal", modules = modules, raw_data = raw_data, filter = filter)
-    logger::log_trace("srv_teal_with_splash initialized module with data { toString(get_dataname(data))}.")
+    if (!is.reactive(raw_data)) {
+      stop("The delayed loading module has to return a reactive object.")
+    }
+
+    raw_data_checked <- reactive({
+      # custom module can return error
+      data <- tryCatch(raw_data(), error = function(e) e)
+
+      # there is an empty reactive event on init!
+      if (inherits(data, "shiny.silent.error") && identical(data$message, "")) {
+        return(NULL)
+      }
+
+      # to handle qenv.error
+      if (inherits(data, "qenv.error")) {
+        validate(
+          need(
+            FALSE,
+            paste(
+              "Error when executing `teal_data_module`:\n",
+              data$message,
+              "\n Check your inputs or contact app developer if error persist"
+            )
+          )
+        )
+      }
+
+      # to handle module non-qenv errors
+      if (inherits(data, "error")) {
+        validate(
+          need(
+            FALSE,
+            paste0(
+              "Error when executing `teal_data_module`:",
+              attr(data, "condition")$message,
+              "\n Check your inputs or contact app developer if error persist"
+            )
+          )
+        )
+      }
+
+      validate(
+        need(
+          inherits(data, "teal_data"),
+          paste(
+            "Error: `teal_data_module` didn't return `teal_data` object",
+            "\n Check your inputs or contact app developer if error persist"
+          )
+        )
+      )
+
+      is_modules_ok <- check_modules_datanames(modules, teal.data::datanames(data))
+      is_filter_ok <- check_filter_datanames(filter, teal.data::datanames(data))
+
+      validate(need(isTRUE(is_modules_ok), is_modules_ok))
+
+      if (!isTRUE(is_filter_ok)) {
+        showNotification(is_filter_ok, type = "warning")
```

Comment:
The message needs fixing (`NULL NULL`?).
![image](https://github.com/insightsengineering/teal/assets/114988527/fdc3153e-fd56-4dbc-a46e-4a0755ebd2b2)

I still think this shows up for too short a time, I can bearly read it with a single message, let alone with multiple.
![image](https://github.com/insightsengineering/teal/assets/114988527/c61b2820-8b3b-4e6b-957c-667e96986aaa)

Reply:
5 seconds to 10 seconds?

Reply:
Well...

![image](https://github.com/insightsengineering/teal/assets/114988527/474a6f21-b661-485d-a2fa-aa873da6eadd)


You be the judge.

Reply:
Maybe: mistake of app developer shouldn't be show to app user at all. I'll just write "some filters were not applied because of incompatibility with data" and the details will be in logs

Reply:
I think that might be better. BUT. What if a remote data source drops a dataset between app sessions?
App dev builds app with downloaded data and with filters for `dataset1`, `dataset2`, `dataset3`.
App works.
Database removes `dataset3` for whatever reason.
App fails.

Is that possible?

On the other hand, maybe that is something the app dev should handle.